### ISSUE TITLE
sqlfluff not catching trailing comma error in UPDATE

### ISSUE THREAD

**Issue by ghodss** [sqlfluff/sqlfluff#5598]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I am fairly certain I am missing or misunderstanding something basic. See below.

### Observed & Expected Behaviour

```console
$ cat querytest.sql
INSERT INTO users (
    id, name
) VALUES (
    $1, $2
)
RETURNING *;

UPDATE users
SET
    id = $1,
    name = $2, -- invalid trailing comma here
WHERE id = $1;
$ sqlfluff lint --dialect postgres querytest.sql
All Finished 📜 🎉!
$ # why?
```

I would expect an error for the UPDATE statement, but I don't see one here. I do see one for INSERT:

```console
$ cat querytest.sql
INSERT INTO users (
    id, name
) VALUES (
    $1, $2, -- invalid trailing comma here
)
RETURNING *;

UPDATE users
SET
    id = $1,
    name = $2
WHERE id = $1;
$ sqlfluff lint --dialect postgres querytest.sql
== [querytest.sql] FAIL
L:   4 | P:  11 |  PRS | Line 4, Position 11: Found unparsable section: ','
WARNING: Parsing errors found and dialect is set to 'postgres'. Have you configured your dialect correctly?
All Finished 📜 🎉!
$ # as expected
```

If I try to use the UPDATE statement with the trailing comma against Postgres, I get: "syntax error at or near "WHERE"", as I would expect, so it's definitely not valid. Also I tried without the inline SQL comments with the same result.

What am I missing?

### Dialect

Postgres

### Version

```
$ sqlfluff --version
sqlfluff, version 2.3.5
```

### Configuration

None.

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by peterbud** [comment]
Hi @ghodss  - thank you for opening this issue.

WIth the latest `sqlfluff` you get an unparsable error with the example statement:
```sql
INSERT INTO users (
    id, name
) VALUES (
    $1, $2
)
RETURNING *;

UPDATE users
SET
    id = $1,
    name = $2, -- invalid trailing comma here
WHERE id = $1;
```

```bash
...
==== parsing violations ====
L:  11 | P:  14 |  PRS | Line 11, Position 14: Found unparsable section: ', -- invalid
                       | trailing comma here\nWHERE i...'
```
